packages/ssz/src/types/composite/container.ts

```diff
@@ -246,9 +249,9 @@ export class ContainerType<T extends ObjectLike = ObjectLike> extends CompositeT
       throw new Error("Invalid JSON container: expected Object");
     }
     const value = {} as T;
+    const expectedCase = options && options.case;
+    const customCasingMap = expectedCase === "declared" ? this.casingMap : options && options.casingMap;
```

Comment:
if `this.casingMap` exists, then ignore all json options and always apply that mapping